Add support for local debugging lambda in Rider

sdubov opened the pull request:
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
- Implement DotNet debugger support
- Update gradle task to download Rider Debugger Worker process runner
- Update SAM debugger logic to async calls when init debugger process

## Motivation and Context
Used should be able to run Lambda functions under debug to handle errors locally before deploying it to production.

## Related Issue(s)
None

## Testing
Environment:
```
JetBrains Rider 2019.2 EAP
Runtime version: 1.8.0_212-release-1566-b8 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.5
GC: PS Scavenge, PS MarkSweep
Memory: 480M
Cores: 8
```
- Checked debugger is initialized and is started
- Checked debugger can be stopped
- Checked docker container is stopped and terminated on debugger stop

## Screenshots (if appropriate)
<img width="1138" alt="RiderLambdaLocalDebug" src="https://user-images.githubusercontent.com/6064345/60173907-83d62f80-9818-11e9-9706-212d22297343.png">

## Checklist
- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

sdubov committed:
Add support for local debugging lambda in Rider

- Implement DotNet debugger support
- Update gradle task to download Rider Debugger Worker process runner
- Update SAM debugger logic to async calls when init debugger process

korifey commented on jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamDebugger.kt:
```diff
@@ -28,12 +36,37 @@ internal class SamDebugger : SamRunner() {
+
+        ProgressManager.getInstance().run(object : Task.Backgroundable(environment.project, message("lambda.debug.waiting"), false) {
+            override fun run(indicator: ProgressIndicator) {
+                spinUntil(debugSupport.debuggerAttachTimeoutMs) { isDebuggerAttachDone }
```
Here you can check return value of spinUnit. In case of 'false' maybe it worth to log a problem

sdubov replied:
Fixed.

abrooksv committed:
Merge branch 'feature/rider-support' into sdubov-lambda-debug-local

sdubov commented:
Fixed. Closing the PR since it is merged to upstream.

sdubov committed:
Fixes from code review

- Add JetBrains.Rider.Debugger.Launcher app to Rider SDK
- Fix debugger initialization logic

sdubov committed:
Remove redundant imports and constants in gradle build file

sdubov committed:
Move check for running Docker from DotNet to SamDebug base class
